### PR TITLE
Integrate 'Extended AppArmor interface trace test' into openQA

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://progress.opensuse.org/issues/111242

Verification run:
 https://openqa.suse.de/tests/8788184# (s390x)
 https://openqa.suse.de/tests/8788187# (aarch64)
 https://openqa.suse.de/tests/8788188# (64bit)

### CODE REVIEW

File: tests/security/cc/extended_apparmor_interface_trace_test.pm [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+sub run {
+    my ($self) = shift;
+
+    select_console 'root-console';
+
+    zypper_call('in strace');
+
+    assert_script_run("cd $audit_test::test_dir/audit-test/kvm_svirt_apparmor/tests");
+
+    # Cleanup the log files which are generated by last test
```
Comment (rfan1) [inline tests/security/cc/extended_apparmor_interface_trace_test.pm:26]: Clean up the previous test logs

File: tests/security/cc/extended_apparmor_interface_trace_test.pm [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    script_run('rm -rf /tmp/vm-sep/');
+
+    # Record the cmdline information
```
Comment (rfan1) [inline tests/security/cc/extended_apparmor_interface_trace_test.pm:29]: "kernel parameters  during boot"
@Amrysliu I suggest you record_info here :)

File: tests/security/cc/extended_apparmor_interface_trace_test.pm [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    my $strace_log = 'strace.log';
+    assert_script_run("sed -i 's/\\/sbin\\/apparmor_parser -a \$C_LABEL/strace -o $strace_log &/' $test_file");
+
+    # Run the test script with strace
+    script_run("./$test_file", timeout => 120);
+
+    # Check if the strace log is generated
+    assert_script_run("test -e $strace_log");
+
+    if (script_run("cat $strace_log | grep '\\.load'") != 0) {
```
Comment (rfan1) [inline tests/security/cc/extended_apparmor_interface_trace_test.pm:43]: Minor comment:

grep ''\\.load' $strace_log :)

File: tests/security/cc/extended_apparmor_interface_trace_test.pm [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    # Run the test script with strace -tt -ff
+    assert_script_run("sed -i 's/strace -o/strace -tt -ff -o/' $test_file");
+
+    # Run the test script with strace -ff
+    script_run("./$test_file", timeout => 120);
+
+    # Check if the strace logs with pid are generated
+    assert_script_run('find . -name "strace.log.*"');
+
+    # strace option -ff was used to ensure that strace also follows children,
```
Comment (rfan1) [inline tests/security/cc/extended_apparmor_interface_trace_test.pm:56]: Use strace option -ff to

File: tests/security/cc/extended_apparmor_interface_trace_test.pm (hunk 3 above)
Comment (Amrysliu) [reply tests/security/cc/extended_apparmor_interface_trace_test.pm]: fixed

**Review by rfan1 — APPROVED** [review]
LGTM

**Review by lilyeyes — APPROVED** [review]
LGTM.

**Pull request merged**